samples/sensor/ism330dhcx/src/main.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022 Yestin Sun
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/sensor.h>
+
+#define ISM330DHCX_ODR_12HZ5	12
+
+static inline float out_ev(struct sensor_value *val)
+{
+	return (val->val1 + (float)val->val2 / 1000000);
+}
+
+static void fetch_and_display(const struct device *dev)
+{
+	struct sensor_value x, y, z;
+	static int trig_cnt;
+
+	trig_cnt++;
+
+	/* ism330dhcx accel */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_ACCEL_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_X, &x);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_Y, &y);
+	sensor_channel_get(dev, SENSOR_CHAN_ACCEL_Z, &z);
+
+	printf("accel x:%f m/s2 y:%f m/s2 z:%f m/s2\n",
+			out_ev(&x), out_ev(&y), out_ev(&z));
+
+	/* ism330dhcx gyro */
+	sensor_sample_fetch_chan(dev, SENSOR_CHAN_GYRO_XYZ);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_X, &x);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_Y, &y);
+	sensor_channel_get(dev, SENSOR_CHAN_GYRO_Z, &z);
+
+	printf("gyro x:%f dps y:%f dps z:%f dps\n",
+			out_ev(&x), out_ev(&y), out_ev(&z));
+
+	printf("trig_cnt:%d\n\n", trig_cnt);
+}
+
+static int set_sampling_freq(const struct device *dev)
+{
+	int ret = 0;
+	struct sensor_value odr_attr;
+
+	/* set accel/gyro sampling frequency to 12.5 Hz */
+	odr_attr.val1 = ISM330DHCX_ODR_12HZ5;
```

Comment:
`ISM330DHCX_ODR_12HZ5` definition is not needed, sensor_value has associated units (it's setting sampling frequency).

Reply:
Could you please elaborate more on what do you mean that sensor_value has associated units?

The ISM330DHCX_ODR_12HZ5 definition used here is to indicate that it is setting the sampling frequency to 12.5HZ, but the actual value assigned is 12.